pkg/workceptor/kubernetes.go

```diff
@@ -338,115 +343,174 @@ func (kw *KubeUnit) KubeLoggingWithReconnect(streamWait *sync.WaitGroup, stdout
 				"Error getting pod %s/%s. Will retry %d more times. Error: %s",
 				podNamespace,
 				podName,
-				retries,
+				retryGetPod,
 				err,
 			)
-			time.Sleep(time.Second)
+			time.Sleep(time.Second * time.Duration(curPodDelay))
+			prevPodDelay, curPodDelay = curPodDelay, prevPodDelay+curPodDelay
 		}
 		if err != nil {
-			errMsg := fmt.Sprintf("Error getting pod %s/%s. Error: %s", podNamespace, podName, err)
-			kw.GetWorkceptor().nc.GetLogger().Error(errMsg) //nolint:govet
-			kw.UpdateBasicStatus(WorkStateFailed, errMsg, 0)
+			errMsg := fmt.Errorf("Error getting pod %s/%s. Error: %s", podNamespace, podName, err)
+			kw.GetWorkceptor().nc.GetLogger().Error("%s", errMsg.Error())
+			*stdoutErr = errMsg
 
 			// fail to get pod, no need to continue
 			return
 		}
+		prevPodDelay, curPodDelay = 1, 1
+
+		// Reset successfulWrite on each reconnection attempt to ensure proper duplicate detection
+		successfulWrite = false
 
 		logStream, err := kw.kubeLoggingConnectionHandler(true, sinceTime)
 		if err != nil {
 			// fail to get log stream, no need to continue
 			return
 		}
+		defer logStream.Close()
 
 		// read from logstream
 		streamReader := bufio.NewReader(logStream)
-		for *stdinErr == nil { // check between every line read to see if we need to stop reading
+		for { // check between every line read to see if we need to stop reading
 			line, err := streamReader.ReadString('\n')
 			if err != nil {
-				if kw.GetContext().Err() == context.Canceled {
-					kw.GetWorkceptor().nc.GetLogger().Info(
-						"Context was canceled while reading logs for pod %s/%s. Assuming pod has finished",
+				// First check if the error is not EOF, if error is not EOF retry 5 times if error persists set error and mark the job as failed.
+				if err != io.EOF {
+					retryGetLogStream--
+					if retryGetLogStream > 0 {
+						kw.GetWorkceptor().nc.GetLogger().Info(
+							"Detected Error: %s for pod %s/%s that is in a running state. Will retry %d more times.",
```

Comment:
```suggestion
							"Detected non-EOF Error: %s for pod %s/%s. Will retry %d more times.",
```

At this point we haven't checked that the pod (or container) is in a running state, so I think logging that would be confusing. I suggest we report that this is a non-EOF error.